.github/workflows/main.yaml

```diff
@@ -23,24 +9,45 @@ on:
       - 'master'
       - 'staging'
       - 'trying'
+    tags:
+      - '[0-9]+.[0-9]+.[0-9]+*'
 jobs:
+  setup:
+    name: Set up
+    runs-on: ubuntu-latest
+    outputs:
+      VERSION: ${{ steps.setup.outputs.VERSION }}
+      DOING_RELEASE: ${{ steps.setup.outputs.DOING_RELEASE }}
+    steps:
+      - name: Set up env vars
+        id: setup
+        shell: bash
+        run: |
+          VERSION=${GITHUB_REF/refs\/tags\//}
+          echo ::set-output name=VERSION::${VERSION}
+          DOING_RELEASE=$(echo $VERSION | grep -q '[0-9]\+\.[0-9]\+\.[0-9]\+\(-.*\)\?' && echo -n '1' || echo -n '0')
```

Comment:
Maybe
```suggestion
          DOING_RELEASE=$(echo $VERSION | grep -cq '[0-9]\+\.[0-9]\+\.[0-9]\+\(-.*\)\?')
```
?

Reply:
We need the result as a string I think, I'm not 100% sure.

This line was a nightmare to get working correctly

Reply:
Yeah, the env var isn't set if I do `-cq` with no echo